src/astro/utils/table_handler_new.py

```diff
@@ -0,0 +1,94 @@
+import inspect
+from abc import ABC
+from typing import Callable, Optional, Tuple
+
+from astro.databases import create_database
+from astro.sql.table import Metadata, Table
+
+
+class TableHandler(ABC):
```

Comment:
Do we need a class for this? Could we just have two functions with the shared functionality?